File: WalletTracker/app/src/main/java/com/chtima/wallettracker/dao/UserDao.java
package com.chtima.wallettracker.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.chtima.wallettracker.models.User;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;
import io.reactivex.rxjava3.core.Maybe;
import io.reactivex.rxjava3.core.Single;

@Dao
public interface UserDao {

    @Insert
    Maybe<Long> insert (User user);

    @Query("SELECT * FROM users LIMIT 1")
    Single<User> getFirst ();

    @Query("SELECT * FROM users")
    Flowable<List<User>> getUsers();

    @Update
    Completable update(User user);

}

File: WalletTracker/app/src/main/java/com/chtima/wallettracker/db/dao/UserDao.kt
package com.chtima.wallettracker.db.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.chtima.wallettracker.models.User
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Flowable
import io.reactivex.rxjava3.core.Maybe
import io.reactivex.rxjava3.core.Single

@Dao
interface UserDao {

    @Insert
    fun insert(user: User): Maybe<Long>

    @Query("SELECT * FROM users")
    fun getUsers(): Flowable<List<User>>

    @Query("SELECT * FROM users WHERE id = :id LIMIT 1")
    fun getUserById(id: Long): Single<User>

    @Update
    fun update(user: User): Completable

}